Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m

```diff
@@ -1126,17 +1280,103 @@ - (int)maxMessageWidthForContentWidth:(int)contentWidth
     return (int)floor(contentWidth * 0.8f);
 }
 
+- (CGSize)quotedMessageSizeForViewWidth:(int)viewWidth
+                           contentWidth:(int)contentWidth
+                         includeMargins:(BOOL)includeMargins
+{
+    OWSAssert(self.viewItem);
+    OWSAssert([self.viewItem.interaction isKindOfClass:[TSMessage class]]);
+
+    CGSize result = CGSizeZero;
+
+    if (!self.isQuotedReply) {
+        return result;
+    }
+
+    result.width += self.quotedMessageHInset;
+    result.width += self.quotedReplyStripeThickness;
+    result.width += self.quotedReplyStripeHSpacing;
+
+    result.height += self.quotedMessageTopInset;
+
+    CGFloat thumbnailHeight = 0.f;
+    if (self.hasQuotedAttachmentThumbnail) {
+        result.width += self.quotedThumbnailHSpacing;
+        result.width += self.quotedThumbnailSize;
+
+        thumbnailHeight = self.quotedThumbnailSize;
+    } else {
+        result.width += self.quotedContentTrailingMargin;
+    }
+
+    result.width += self.quotedMessageHInset;
+
+    // Once we've determined everything _except_ the size of the text
+    // content (i.e. the quoted author and the quoted text (if any)),
+    // we can determine the size of the text content.
+    const int maxMessageWidth = [self maxMessageWidthForContentWidth:contentWidth];
+    CGFloat maxTextWidth = (maxMessageWidth - (self.textTrailingMargin + self.textLeadingMargin + result.width));
+    CGFloat textWidth = 0.f;
+
+    // Author
+    {
+        OWSContactsManager *contactsManager = Environment.current.contactsManager;
+        NSString *quotedAuthor = [contactsManager displayNameForPhoneIdentifier:self.viewItem.quotedRecipientId];
+
+        UILabel *quotedAuthorLabel = [UILabel new];
```

Comment:
For perf, we might want to cache a pre-configured sizing label.

Reply:
I suspect it'll be cheaper not to since quoted replies will be less common.